py/_path/local.py

```diff
@@ -820,14 +820,16 @@ def parse_num(path):
                 except ValueError:
                     pass
 
+        exclusive_flag = 'x' if not sys.platform.startswith('win') else ''
+
         def create_lockfile(path):
             """ exclusively create lockfile. Throws when failed """
             mypid = os.getpid()
             lockfile = path.join('.lock')
             if hasattr(lockfile, 'mksymlinkto'):
                 lockfile.mksymlinkto(str(mypid))
             else:
-                lockfile.write(str(mypid), 'wx')
+                lockfile.write(str(mypid), 'w' + exclusive_flag)
```

Comment:
as far as i can tell without the exclusive flag the lockfile creation is broken

Reply:
Not sure what to do here, because passing `wx` to `lockfile.write` raises an exception on Windows:

```
ValueError: must have exactly one of create/read/write/append mode
```

Reply:
Also, it is problematic then that removing this flag didn't break *any* test. Either it is not necessary or we are not covering the case where the flag is needed.

Reply:
the x flag is absolutely critical as it ensures exclusive open - else the operation wont fail if the element exists

Reply:
> the x flag is absolutely critical as it ensures exclusive open - else the operation wont fail if the element exists

I got that. 😄 It is still a problem that no test breaks on Windows after the flag has been removed.

Just to be clear: Windows does not support the `x` flag so we can't publish `1.5` with this change otherwise we will break every new pytest installation where the test suite uses `tmpdir`.

How should we proceed?

Reply:
i ccd you in the relevant pr

Reply:
Thanks! Let's move the discussion there then.